realm/src/androidTest/java/io/realm/RealmJsonTest.java

```diff
@@ -452,6 +457,15 @@ public void testCreateOrUpdateObjectStream_noPrimaryKeyThrows() throws IOExcepti
         fail();
     }
 
+    public void testCreateOrUpdateObjectStream_invalidJSonThrows() throws IOException {
```

Comment:
Actually they don't. Junit3 will still report an error in that case (I just tested it). Perhaps you are thinking about Junit4 and its expected annotation?